internal/timerange/timerange.go

```diff
@@ -0,0 +1,23 @@
+package timerange
+
+import (
+	"errors"
+	"time"
+)
+
+type Range struct {
+	Start *time.Time
+	End   *time.Time
+}
+
+func New(start, end *time.Time) (Range, error) {
```

Comment:
Why are pointers used for time.Time instead of values?

Reply:
@sevein I'm using nil values to represent half-open ranges - a nil start or end time represents any time.  So a range with a start time of "2024-09-25 15:15:25 UTC" and a nil end time would represent any time equal to or after "2024-09-25 15:15:25 UTC".

Passing pointers to the `New()` constructor is awkward though, and I would like to improve the ergonomics. I suppose a half open range could be approximated by choosing a very early start time or a very late end time instead of using pointers, and using values would simplify things.